README.md

```diff
@@ -1 +1,415 @@
+
+### Top-level Arguments
+
+#### Module Configuration
+
+- **`module_enabled`**: _(Optional `bool`)_
+
+  Specifies whether resources in the module will be created.
+  Default is `true`.
+
+- **`module_depends_on`**: _(Optional `list(dependencies)`)_
+
+  A list of dependencies. Any object can be _assigned_ to this list to define a hidden external dependency.
+
+  Example:
+  ```hcl
+  module_depends_on = [
+    google_network.network
+  ]
+  ```
+
+#### Main Resource Configuration
+
+- **`dataset_id`**: **_(Required `string`)_**
+
+  A unique ID for this dataset, without the project name. The ID must contain only letters `(a-z, A-Z)`, numbers `(0-9)`, or underscores `(_)`. The maximum length is `,024 characters.
+
+- **`friendly_name`**: _(Optional `string`)_
+
+  Friendly name for the dataset being provisioned."
+  Default is `null`.
+
+- **`description`**: _(Optional `string`)_
+
+  A user-friendly description of the dataset.
+  Default is `null`.
+
+- **`project`**: _(Optional `string`)_
+
+  The ID of the project in which the resource belongs. If it is not provided, the provider project is used."
+  Default is `null`.
+
+- **`location`**: _(Optional `string`)_
+
+  The geographic location where the dataset should reside.
+  Default is `null`.
+
+- **`default_table_expiration_ms`**: _(Optional `number`)_
+
+  The default lifetime of all tables in the dataset, in milliseconds.
+  Default is `null`.
+
+- **`default_partition_expiration_ms`**: _(Optional `number`)_
+
+  The default partition expiration for all partitioned tables in the dataset, in milliseconds.
+  Default is `null`.
+
+- **`labels`**: _(Optional `map(string)`)_
+
+  Key value pairs in a map for dataset labels.
+  Default is `{}`.
+
+- **`access`**: _(Optional `list(access)`)_
+
+  An array of objects that define dataset access for one or more entities.
+  Default is `[]`.
+
+  An `access` object can have the following fields:
+
+  - **`domain`**: _(Optional `string`)_
+
+    A domain to grant access to. Any users signed in with the domain specified will be granted the specified access.
+    Default is `null`.
+
+  - **`role`**: _(Optional `string`)_
+
+    Describes the rights granted to the user specified by the other member of the access object. Basic, predefined, and custom roles are supported.
+    Default is `null`.
+
+  - **`group_by_email`**: _(Optional `string`)_
+
+    An email address of a Google Group to grant access to.
+    Default is `null`.
+
+  - **`user_by_email`**: _(Optional `string`)_
+
+    An email address of a Google Group to grant access to.
+    Default is `null`.
+
+  - **`special_group`**: _(Optional `string`)_
+
+    A special group to grant access to. Possible values include:
+    projectOwners: Owners of the enclosing project.
+    ProjectReaders: Readers of the enclosing project.
+    projectWriters: Writers of the enclosing project.
+    allAuthenticatedUsers: All authenticated BigQuery users.
+    Default is `null`.
```

Comment:
please format to make it readable.. also capitalization seems broken.. what is the actual name?